### PR TITLE
Add `build_to` Builder config option

### PR DESCRIPTION
**Pull request by natebosch**
Towards #647

The value is defaulted based on the `auto_apply` option, since for
certain values the only possibility is `True`.

### CODE REVIEW

**Comment by natebosch** [comment]
`build_to_cache`?

**Comment by grouma** [comment]
Doesn't it always populate the cache? If not, the build_to_cache works.

**Comment by natebosch** [comment]
We now support mixing builders which write to the cache (angular, ddc) with builders that write directly to the source tree (built_value, etc). Only the files which are generated end up in the build cache, we don't write source files there.

I'm a little torn on the naming. I agree that `build_to_cache` is a more direct map to what will happen. But `hide_output` is, I think, a more direct map to the goal the builder author has...

**Comment by natebosch** [comment]
After a quick discussion I think I might change this to an enum-like value with two choices:

`build_to: source` and `build_to: cache`

Thoughts?

**Comment by grouma** [comment]
I prefer the enum approach.

**Comment by jakemac53** [comment]
I like the enum approach as well

**Comment by natebosch** [comment]
Made it an enum `BuildTo` with options `source` and `cache`. PTAL

File: build_config/README.md [hunk 1]
```diff
@@ -40,6 +40,11 @@ the following keys:
 - **is_optional**: Optional, boolean. Specifies whether a Builder can be run
   lazily, such that it won't execute until one of it's outputs is requested by a
   later Builder. This option should be rare. Defaults to `False`.
+- **hide_output**: Optional, boolean. Specifies whether the outputs produced by
```
Comment (jakemac53) [inline build_config/README.md]: update this note and chanegelog

**Pull request merged**